### PR TITLE
added checkers and imports

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

**Review by costezki — APPROVED** [review]

File: src/common/fetchers.xsl [hunk 1]
```diff
@@ -37,6 +37,19 @@
         <xsl:sequence select="$root//elements/element[@name=$name]"/>
     </xsl:function>
     
+    <xd:doc>
+        <xd:desc>fetch the element attribute with a given name</xd:desc>
```
Comment (costezki) [inline src/common/fetchers.xsl:41]: Surprisingly, we did not have this before. 

don't we check the multiple usage of attributes like we check multipel usage of connectors? 
If not then please signal this to Csongor and implement accordingly in the Convention's checking report.

File: src/html-conventions-lib/association-html-conventions.xsl [hunk 2]
```diff
@@ -113,6 +115,13 @@
                 <xsl:call-template name="associationSourceTargetTypes">
                     <xsl:with-param name="connector" select="."/>
                 </xsl:call-template>
+                <xsl:call-template name="connectorUniqueName">
```
Comment (costezki) [inline src/html-conventions-lib/association-html-conventions.xsl:118]: I like how you started breaking down the functionality into smaller sub-functions. This is professional identification and separation of concerns.

File: src/html-conventions-lib/class-attributes-html-conventions.xsl [hunk 3]
```diff
@@ -89,6 +89,10 @@
             <xsl:call-template name="namePlural">
                 <xsl:with-param name="element" select="."/>
             </xsl:call-template>
+            <xsl:call-template name="elementUniqueName">
+                <xsl:with-param name="element" select="."/>
+                <xsl:with-param name="isAttribute" select="fn:true()"/>
```
Comment (costezki) [inline src/html-conventions-lib/class-attributes-html-conventions.xsl:94]: What "isAttribute"? The variable name is not clear.

File: src/html-conventions-lib/class-html-conventions.xsl [hunk 4]
```diff
@@ -90,12 +90,13 @@
             <xsl:call-template name="nonPublicElement">
                 <xsl:with-param name="element" select="."/>
             </xsl:call-template>
+            <xsl:call-template name="elementUniqueName">
+                <xsl:with-param name="element" select="."/>
+                <xsl:with-param name="isAttribute" select="fn:false()"/>
```
Comment (costezki) [inline src/html-conventions-lib/class-html-conventions.xsl:95]: idem;

File: src/html-conventions-lib/class-html-conventions.xsl [hunk 5]
```diff
@@ -215,7 +216,7 @@
 
  
 
-    <xd:doc>
+<!--    <xd:doc>
```
Comment (costezki) [inline src/html-conventions-lib/class-html-conventions.xsl]: How come this is commented out? What does this mean? 

If we do not need this functionality, then delete it. But judging by the documentation, it seems relevant.

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 6]
```diff
@@ -17,7 +17,8 @@
     <xsl:import href="../common/checkers.xsl"/>
     <xsl:import href="utils-html-conventions.xsl"/>
 
-    <xsl:variable name="elementTypes" select="('class', 'enumeration', 'dataType', 'package', 'object')"/>
+    <xsl:variable name="elementTypes"
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl:20]: Should we have a similar list for connector types? 

Feel free to implement a new checker (after the release) for signalling presence of non relevant element and connector types. Report this to Csongor.

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 7]
```diff
@@ -202,7 +203,7 @@
         <xsl:variable name="elementName" select="$element/@name"/>
         <xsl:variable name="noElementDescription"
             select="
-            if ($elementType = $elementTypes) then
+                if ($elementType = $elementTypes) then
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl:206]: Separate checking the element type from getting its documentation. 

Checking "type" shall be applied everywhere, or nowhere, when getting a XML sub-element/attribute (e.g. documentation).

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 8]
```diff
@@ -256,7 +257,7 @@
         <xsl:param name="elementType"/>
         <xsl:variable name="isStereotypeValid"
             select="
-            if ($elementType = $elementTypes) then
+                if ($elementType = $elementTypes) then
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl:260]: We shall check type once, and only once in a function/template

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 9]
```diff
@@ -369,21 +370,71 @@
...
+            concept names. </xd:desc>
+        <xd:param name="element"/>
+        <xd:param name="isAttribute"/>
+    </xd:doc>
+    <xsl:template name="elementUniqueName">
+        <xsl:param name="element"/>
+        <xsl:param name="isAttribute"/>
+        <xsl:if test="boolean($element/@name)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($element/@name, root($element))"/>
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl:403]: Here, for example we do not check type (so we might be getting non-permitted element types, e.g. uml:Component, or other ones, which would not be correct).

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 10]
```diff
@@ -369,21 +370,71 @@
...
+            <xsl:variable name="connectorsFound"
+                select="f:getConnectorByName($element/@name, root($element))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($element/@name, root($element))"/>
+            <xsl:choose>
+                <xsl:when test="$isAttribute = fn:true()">
+                    <xsl:sequence
+                        select="
+                            if (count($elementsFound) > 1 or count($connectorsFound) > 0) then
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl]: The attributes CAN be repeated across classes, but they cannot be names like any of the connectors or classes. The logic/variable naming looks weird. Please revise it. 

It is confusing why among elements it shall be > 1.

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 11]
```diff
@@ -369,21 +370,71 @@
...
-            f:generateHtmlWarning(fn:concat('The element ', $element/@name, ' is non-public. All elements shall be public '))"
+                if ($elementScope = 'public') then
+                    ()
+                else
+                    f:generateHtmlWarning(fn:concat('The element ', $element/@name, ' is non-public. All elements shall be public '))"
         />
     </xsl:template>
 
+    <xd:doc>
+        <xd:desc>[common-name-unique] - The name $value$ is not unique. The Concept names should be
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl:392]: We need 3 different logics here. Its si better and cleaner to split this template/function into 3 ones, one for elements, one for attributes, one for connectors. 

What do you think ?

File: src/html-conventions-lib/common-elements-html-conventions.xsl [hunk 12]
```diff
@@ -369,21 +370,71 @@
...
+                                ()
+                            
+                            "
+                    />
+                </xsl:otherwise>
+            </xsl:choose>
+
+        </xsl:if>
+    </xsl:template>
+
```
Comment (costezki) [inline src/html-conventions-lib/common-elements-html-conventions.xsl:439]: looks incomplete

File: src/html-conventions-lib/data-type-html-conventions.xsl [hunk 13]
```diff
@@ -90,15 +90,20 @@
             <xsl:call-template name="nonPublicElement">
                 <xsl:with-param name="element" select="."/>
             </xsl:call-template>
+            
+            <xsl:call-template name="elementUniqueName">
+                <xsl:with-param name="element" select="."/>
+                <xsl:with-param name="isAttribute" select="fn:false()"/>
```
Comment (costezki) [inline src/html-conventions-lib/data-type-html-conventions.xsl:96]: Eliminate this "isAttribute" parameter by breaking up the logic into  3 different functions.

File: src/html-conventions-lib/data-type-html-conventions.xsl [hunk 14]
```diff
@@ -90,15 +90,20 @@
...
+            </xsl:call-template>
             <!--    End of common checkers rules     -->   
             <!--    Start of specific checker rules-->
             
             <xsl:call-template name="dataTypeIncorrectType">
                 <xsl:with-param name="dataTypeElement" select="."/>
             </xsl:call-template>
-            <xsl:call-template name="dataTypeUniqueName">
+<!--            <xsl:call-template name="dataTypeUniqueName">
```
Comment (costezki) [inline src/html-conventions-lib/data-type-html-conventions.xsl]: delete comments if they are no longer needed

File: src/html-conventions-lib/data-type-html-conventions.xsl [hunk 15]
```diff
@@ -137,7 +142,7 @@
     </xsl:template>
     
     
-    <xd:doc>
+<!--    <xd:doc>
```
Comment (costezki) [inline src/html-conventions-lib/data-type-html-conventions.xsl]: why is this commented out? do we need it?

File: src/html-conventions-lib/dependency-html-conventions.xsl [hunk 16]
```diff
@@ -105,6 +105,13 @@
                     <xsl:call-template name="dependencySourceTargetTypes">
                         <xsl:with-param name="dependencyConnector" select="."/>
                     </xsl:call-template>
+                    <xsl:call-template name="connectorUniqueName">
+                        <xsl:with-param name="connector" select="."/>
+                    </xsl:call-template> 
+                    <xsl:call-template name="connectorRoleCheck">
+                        <xsl:with-param name="connector" select="."/>
+                        <xsl:with-param name="isDependency" select="fn:true()"/>
```
Comment (costezki) [inline src/html-conventions-lib/dependency-html-conventions.xsl:113]: Looks like you have two logics in one function. 
Please get rid of the "isDependecy" flag and break down the logic into two sub-functionalities.

File: src/html-conventions-lib/enumeration-html-conventions.xsl [hunk 17]
```diff
@@ -88,16 +88,20 @@
             <xsl:call-template name="nonPublicElement">
                 <xsl:with-param name="element" select="."/>
             </xsl:call-template>
+            <xsl:call-template name="elementUniqueName">
+                <xsl:with-param name="element" select="."/>
+                <xsl:with-param name="isAttribute" select="fn:false()"/>
```
Comment (costezki) [inline src/html-conventions-lib/enumeration-html-conventions.xsl:93]: idem.

File: src/html-conventions-lib/enumeration-html-conventions.xsl [hunk 18]
```diff
@@ -88,16 +88,20 @@
...
+            </xsl:call-template>
             <!--    End of common checkers rules     -->   
             <!--    Start of specific checker rules-->
 
             <xsl:call-template name="enumerationItemsChecker">
                 <xsl:with-param name="enumeration" select="."/>
             </xsl:call-template>
 
-            <xsl:call-template name="enumerationUniqueName">
+  <!--          <xsl:call-template name="enumerationUniqueName">
```
Comment (costezki) [inline src/html-conventions-lib/enumeration-html-conventions.xsl]: do we need this?

File: src/html-conventions-lib/enumeration-html-conventions.xsl [hunk 19]
```diff
@@ -142,7 +146,7 @@
 
   
 
-    <xd:doc>
+<!--    <xd:doc>
```
Comment (costezki) [inline src/html-conventions-lib/enumeration-html-conventions.xsl]: delete or uncomment

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 20]
```diff
@@ -501,7 +501,7 @@
         <xsl:variable name="targetName" select="$connector/target/role/@name"/>
         <xsl:sequence
             select="
-            if (($numberOfTags > 0) and not(boolean($targetName))) then
+                if (($numberOfTags > 0) and not(boolean($targetName))) then
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl:504]: I don't think this logic is doing what it is supposed to do. 
Just get all tags and then for each tag check if it has a name.

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 21]
```diff
@@ -541,17 +541,17 @@
         <xsl:sequence
             select="
                 if ($numberOfTags > 0) then
-                    f:generateHtmlWarning(fn:concat('The connector ', f:getConnectorName($connector), ' has tag annotations.', 
-            'The connector is not transformed into a property and therefore any tag will be ignored.'))
+                    f:generateHtmlWarning(fn:concat('The connector ', f:getConnectorName($connector), ' has tag annotations.',
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl:544]: Why is this here? What does it mean? Not sure if I understand the logics.

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 22]
```diff
@@ -688,4 +688,133 @@
...
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($sourceRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($sourceRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl]: this logic is strange. Please recheck based on similar comment above.

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 23]
```diff
@@ -688,4 +688,133 @@
...
+        </xsl:if>
+        <xsl:if test="boolean($targetRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($targetRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($targetRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl]: idem, why >1 and >0 ? 

Break down the name uniqueness checking into 3 functionalities.

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 24]
```diff
@@ -688,4 +688,133 @@
...
+                    
+                    "
+            />
+        </xsl:if>
+    </xsl:template>
+
+
+    <xd:doc>
+        <xd:desc>[association-dependency-checker] - Connector role names are used in other
+            connectors </xd:desc>
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl]: "in other connector TYPES", this is what was meant, right?

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 25]
```diff
@@ -688,4 +688,133 @@
...
+        <xsl:param name="connector"/>
+        <xsl:variable name="sourceRole" select="$connector/source/role/@name"/>
+        <xsl:variable name="targetRole" select="$connector/target/role/@name"/>
+
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:choose>
+                <xsl:when test="$isDependency = fn:true()">
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($sourceRole, root($connector))/properties[@ea_type = 'Association']"/>
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl:752]: why when "is dependecy == true" you select "Association"

This logic looks counter unintuitive. Please refactor. the isDependecy flag shows that more tahn one functionality is merged into the function. Break it down.

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 26]
```diff
@@ -688,4 +688,133 @@
...
+                            else
+                                ()
+                            
+                            "
+                    />
+                </xsl:otherwise>
+            </xsl:choose>
+        </xsl:if>
+
+        <xsl:if test="boolean($targetRole)">
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl:781]: what is the difference from the logic above? 
same/similar code seems to be copied.

Comment (costezki) [reply src/html-conventions-lib/general-connectors-html-convention.xsl:781]: source vs target? Cmon that should have been a function called twice instead of the function code copy-pasted. such code unmaintainable. This is how bugs emerge.

File: src/html-conventions-lib/general-connectors-html-convention.xsl [hunk 27]
```diff
@@ -688,4 +688,133 @@
...
+                    />
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($targetRole, root($connector))/properties[@ea_type = 'Dependency']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $targetRole, ' is used in other connectors.'))
```
Comment (costezki) [inline src/html-conventions-lib/general-connectors-html-convention.xsl:804]: Should we not tell where else is the connector name used (Source -> Target Classes)?

File: src/html-conventions-lib/object-html-conventions.xsl [hunk 28]
```diff
@@ -90,21 +90,25 @@
             <xsl:call-template name="nonPublicElement">
                 <xsl:with-param name="element" select="."/>
             </xsl:call-template>
-            
-            <!--    End of common checkers rules     -->   
+            <xsl:call-template name="elementUniqueName">
+                <xsl:with-param name="element" select="."/>
+                <xsl:with-param name="isAttribute" select="fn:false()"/>
```
Comment (costezki) [inline src/html-conventions-lib/object-html-conventions.xsl:95]: "isAttribute" ...

File: src/html-conventions-lib/object-html-conventions.xsl [hunk 29]
```diff
@@ -90,21 +90,25 @@
...
+
+            <!--    End of common checkers rules     -->
             <!--    Start of specific checker rules-->
             <xsl:call-template name="objectItemsChecker">
                 <xsl:with-param name="object" select="."/>
             </xsl:call-template>
-            
-            <xsl:call-template name="objectUniqueName">
+
+<!--            <xsl:call-template name="objectUniqueName">
```
Comment (costezki) [inline src/html-conventions-lib/object-html-conventions.xsl]: delete or uncomment

File: src/html-conventions-lib/object-html-conventions.xsl [hunk 30]
```diff
@@ -145,7 +149,7 @@
         </xsl:choose>
     </xsl:template>
     
-    <xd:doc>
+<!--    <xd:doc>
```
Comment (costezki) [inline src/html-conventions-lib/object-html-conventions.xsl]: idem

File: src/shacl-shapes.xsl [hunk 31]
```diff
@@ -63,6 +63,7 @@
             <xsl:for-each select="$namespacePrefixes/*:prefixes/*:prefix/@importURI">              
                 <owl:imports rdf:resource="{.}"/>
             </xsl:for-each>      
+            <owl:imports rdf:resource="{$coreArtefactURI}"/>
```
Comment (costezki) [inline src/shacl-shapes.xsl:75]: looks like we have the correct variable name.
Here however we shall add also the "restrictions layer artefact".

**Review by costezki — CHANGES_REQUESTED** [review]
Looks good, as a first iteration yet refactoring is needed and some logic revisited.

File: src/common/fetchers.xsl (hunk 1 above)
Comment (Dragos0000) [reply src/common/fetchers.xsl:41]: This is a fetcher for the an element type attribute. The logic for multiple usage is in conventions scripts not helper functions.

File: src/html-conventions-lib/class-attributes-html-conventions.xsl (hunk 3 above)
Comment (Dragos0000) [reply src/html-conventions-lib/class-attributes-html-conventions.xsl:94]: A flag to tell if the element is an attribute as the logic is different. 

Is the element given as a parameter for the template elementUniqueName an attribute ?

File: src/html-conventions-lib/class-html-conventions.xsl (hunk 4 above)
Comment (Dragos0000) [reply src/html-conventions-lib/class-html-conventions.xsl:95]: idem

File: src/html-conventions-lib/class-html-conventions.xsl (hunk 5 above)
Comment (Dragos0000) [reply src/html-conventions-lib/class-html-conventions.xsl]: as seen in the changes for this file this functionality was replaced by the new template elementUniqueName. It is commented out and not deleted just in case someone will want to revert the changes made in this pull request

Comment (Dragos0000) [reply src/html-conventions-lib/class-html-conventions.xsl]: as seen in the changes for this file this functionality was replaced by the new template elementUniqueName. It is commented out and not deleted just in case someone will want to revert the changes made in this pull request

File: src/html-conventions-lib/common-elements-html-conventions.xsl (hunk 6 above)
Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl:20]: this is a global variable used in this file only for functionality. This was reviewed by you before.

File: src/html-conventions-lib/common-elements-html-conventions.xsl (hunk 7 above)
Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl:206]: as different elements have different xpaths, we use this global variable to know what xpath for the documentation we will take. This is a common checker as per documentation. This was reviewed by you before.

File: src/html-conventions-lib/common-elements-html-conventions.xsl (hunk 12 above)
Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl:439]: what looks incomplete?

File: src/html-conventions-lib/data-type-html-conventions.xsl (hunk 13 above)
Comment (Dragos0000) [reply src/html-conventions-lib/data-type-html-conventions.xsl:96]: The unique name checker was a specific checker for certain elements and it was separate until the decision of this being a common checker. At this point the checker was moved in the common checkers file and applied to all elements as requested. For easy readability maintainability and referencing this is one checker that is applied to all elements as one checker (common checkers are inherited). To do that this template needed a flag for attributes as other elements needed to check the name uniqueness against attributes names.

File: src/html-conventions-lib/class-html-conventions.xsl (hunk 5 above)
Comment (Dragos0000) [reply src/html-conventions-lib/class-html-conventions.xsl]: deleted

File: src/html-conventions-lib/common-elements-html-conventions.xsl (hunk 9 above)
Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl:403]: element type parameter is used to know which xpath to take. ie class, package has different xpath to the documentation than an attribute. as seen in the code there is a global variable that is used to identify element type used only for this file and when the template is called that parameter will have a variable like class. There is no relation between this and the value that is coming from the XML.

File: src/html-conventions-lib/common-elements-html-conventions.xsl (hunk 10 above)
Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl]: fixed mistake. Yes it should be >0

Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl]: fixed mistake. Yes it should be >0

File: src/html-conventions-lib/common-elements-html-conventions.xsl (hunk 11 above)
Comment (Dragos0000) [reply src/html-conventions-lib/common-elements-html-conventions.xsl:392]: connectors are not in this file (2 logic's - attribute is still an element in the grant scheme of things. It is inheriting the same common checkers). The file name is suggesting what is inside (common element checkers). I think this should remain as it is because this is a common checker applied to elements (not including connectors).

File: src/html-conventions-lib/data-type-html-conventions.xsl (hunk 15 above)
Comment (Dragos0000) [reply src/html-conventions-lib/data-type-html-conventions.xsl]: as seen in the changes in this pull request this was replaced by the new template elementUniqueName. It is commented out and not deleted just in case someone will want to revert the changes made in this pull request

File: src/html-conventions-lib/data-type-html-conventions.xsl (hunk 14 above)
Comment (Dragos0000) [reply src/html-conventions-lib/data-type-html-conventions.xsl]: as seen in the changes in this pull request this was replaced by the new template elementUniqueName. It is commented out and not deleted just in case someone will want to revert the changes made in this pull request

Comment (Dragos0000) [reply src/html-conventions-lib/data-type-html-conventions.xsl]: as seen in the changes in this pull request this was replaced by the new template elementUniqueName. It is commented out and not deleted just in case someone will want to revert the changes made in this pull request

File: src/html-conventions-lib/enumeration-html-conventions.xsl (hunk 18 above)
Comment (Dragos0000) [reply src/html-conventions-lib/enumeration-html-conventions.xsl]: deleted

File: src/html-conventions-lib/enumeration-html-conventions.xsl (hunk 19 above)
Comment (Dragos0000) [reply src/html-conventions-lib/enumeration-html-conventions.xsl]: deleted

File: src/html-conventions-lib/dependency-html-conventions.xsl (hunk 16 above)
Comment (Dragos0000) [reply src/html-conventions-lib/dependency-html-conventions.xsl:113]: if the documentation will reflect that this is not a common checker will happily break it into 2 specific checkers. For now I think this should stay in the common if the documentation is showing that this checker is inherited. Let me know if this will appear as specific checker

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 20 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl:504]: this rule is checking if the target role has tags but no name. This is not checking if a tag has a name. There is another checker for this

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 21 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl:544]: This is checking if a connector has tags. this is connector-tag-18 checker as per documentation.

Requirement of the checker: Check if connector has tags, raise a warning

Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl:544]: This is checking if a connector has tags. this is connector-tag-18 checker as per documentation.

Requirement of the checker: Check if connector has tags, raise a warning

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 22 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl]: fixed mistake. Yes it should >0

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 23 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl]: fixed mistake. Yes it should >0

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 24 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl]: yes

Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl]: yes

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 25 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl:752]: if the connector given as parameter is a Dependency you will want to look at Associations if they use the role name from the given Dependency. If this will be a specific checker I will break it down. TBD. Let me know the decision

File: src/shacl-shapes.xsl (hunk 31 above)
Comment (Dragos0000) [reply src/shacl-shapes.xsl:75]: added

File: src/html-conventions-lib/object-html-conventions.xsl (hunk 30 above)
Comment (Dragos0000) [reply src/html-conventions-lib/object-html-conventions.xsl]: done

File: src/html-conventions-lib/object-html-conventions.xsl (hunk 29 above)
Comment (Dragos0000) [reply src/html-conventions-lib/object-html-conventions.xsl]: done

File: src/html-conventions-lib/object-html-conventions.xsl (hunk 28 above)
Comment (Dragos0000) [reply src/html-conventions-lib/object-html-conventions.xsl:95]: missing documentation on this checker

File: src/html-conventions-lib/general-connectors-html-convention.xsl (hunk 27 above)
Comment (Dragos0000) [reply src/html-conventions-lib/general-connectors-html-convention.xsl:804]: Not sure. TBD

**Review by costezki — APPROVED** [review]
